### PR TITLE
Stablize name of reference and nested combination of ref, ptr, and array

### PR DESCRIPTION
**Pull request by upsuper**

### CODE REVIEW

**Comment by upsuper** [comment]
I found one last appearance of "ty_id_XXX" in Stylo's binding file with the current master, and that seems to be a reference. This would stablize name of that case.

It also raised my curiosity of what would happen if I put a reference to a pointer, and it seems that would break the serialization there, so I go further and try to handle combination of reference, pointer, and array at the same time.

File: src/ir/ty.rs [hunk 1]
```diff
@@ -269,12 +269,12 @@ impl Type {
 
     /// Checks whether the name looks like an identifier,
     /// i.e. is alphanumeric (including '_') and does not start with a digit.
-    pub fn is_valid_identifier(name: &str) -> bool {
+    fn is_valid_identifier(name: &str) -> bool {
```
Comment (emilio) [inline src/ir/ty.rs]: At this point we can just use `clang::is_valid_identifier`, I guess.

**Pull request merged**